### PR TITLE
refactor Expression tests and add comments

### PR DESCRIPTION
**Pull request by yetongumich**

### CODE REVIEW

File: gtsam/nonlinear/tests/testExpression.cpp [hunk 1]
```diff
@@ -170,29 +181,29 @@ Point3_ p_cam(x, &Pose3::transformTo, p);
...
+// Check dimensions by calling `dims` method.
 TEST(Expression, BinaryDimensions) {
   map<Key, int> actual, expected = map_list_of<Key, int>(1, 6)(2, 3);
   binary::p_cam.dims(actual);
   EXPECT(actual == expected);
 }
 
 /* ************************************************************************* */
-// dimensions
+// Check dimensions by calling `traceSize` method.
```
Comment (dellaert) [inline gtsam/nonlinear/tests/testExpression.cpp:206]: This actually checks the size of the “trace record”, which is where an execution trace is stored.

File: gtsam/nonlinear/tests/testExpression.cpp [hunk 2]
```diff
@@ -454,18 +470,19 @@ TEST(Expression, UnaryOfSum) {
 /* ************************************************************************* */
 TEST(Expression, WeightedSum) {
   const Key key1(42), key2(67);
+  const Point3 point1(1, 0, 0), point2(0, 1, 0);
```
Comment (dellaert) [inline gtsam/nonlinear/tests/testExpression.cpp:473]: Move to where used ...

File: tests/testExpressionFactor.cpp [hunk 3]
```diff
@@ -59,40 +59,42 @@ Point2_ p(2);
...
   EXPECT_DOUBLES_EQUAL(old.error(values), f.error(values), 1e-9);
   EXPECT_LONGS_EQUAL(2, f.dim());
   boost::shared_ptr<GaussianFactor> gf2 = f.linearize(values);
-  EXPECT( assert_equal(*old.linearize(values), *gf2, 1e-9));
+  EXPECT(assert_equal(*old.linearize(values), *gf2, 1e-9));
 }
 
 /* ************************************************************************* */
-// non-zero noise model
+// Test leaf expression with noise model of differnt variance.
```
Comment (dellaert) [inline tests/testExpressionFactor.cpp:76]: Spelling

**Review by dellaert — APPROVED** [review]
Some nits...

File: gtsam/nonlinear/tests/testExpression.cpp [hunk 4]
```diff
@@ -58,22 +58,23 @@ TEST(Expression, Constant) {
...
+// Test the function `createUnknowns` to create many leaves at once.
 TEST(Expression, Leaves) {
   Values values;
-  Point3 somePoint(1, 2, 3);
+  const Point3 somePoint(1, 2, 3);
   values.insert(Symbol('p', 10), somePoint);
-  std::vector<Point3_> points = createUnknowns<Point3>(10, 'p', 1);
-  EXPECT(assert_equal(somePoint, points.back().value(values)));
+  std::vector<Point3_> pointExpressions = createUnknowns<Point3>(10, 'p', 1);
+  EXPECT(assert_equal(somePoint, pointExpressions.back().value(values)));
```
Comment (gchenfc) [inline gtsam/nonlinear/tests/testExpression.cpp:77]: I think google style guide specifies variable names should be snake case

File: gtsam/nonlinear/tests/testExpression.cpp [hunk 5]
```diff
@@ -133,26 +139,31 @@ template<> struct traits<Class> : public internal::VectorSpace<Class> {};
...
 TEST(Expression, NullaryMethod) {
   // Create expression
-  Expression<Class> p(67);
-  Expression<double> norm_(p, &Class::norm);
+  const Key key(67);
+  Expression<Class> classExpression(key);
+
+  // Make expression from a class method, note how it differs from the function
+  // expressions by leading with the class expression in the constructor.
+  Expression<double> norm_(classExpression, &Class::norm);
```
Comment (gchenfc) [inline gtsam/nonlinear/tests/testExpression.cpp:147]: I appreciate this comment :)

**Review by gchenfc — APPROVED** [review]
Cool!

**Comment by varunagrawal** [comment]
@yetongumich please merge the latest `develop` and the tests should all pass.

**Comment by ProfFan** [comment]
@yetongumich Friendly ping :)

**Comment by ProfFan** [comment]
Thanks @jlblancoc  :)

**Pull request merged**